### ISSUE TITLE
.md links are opening a webpage instead of opening a note in the Markdown Preview 

### ISSUE THREAD

**Issue by YAS01-M** [foambubble/foam#648]
- Foam version: 0.13.3
- Platform: Windows
- Issue occur on the [foam template](https://github.com/foambubble/foam-template) repo: No

**Summary**

Hi 
Since the update to Foam v0.13.3 , when i click on any of .md links , they open a vscode webview test instead of a link of the markdown note . This issue is just in markdown preview , .md links still working  with Ctrl+Click in source page. 

screenshot of the issue when i click on [[todo]] note
![issue](https://user-images.githubusercontent.com/84593166/119190907-da386100-ba75-11eb-8965-027bbe685156.JPG)

**Comment by riccardoferretti** [comment]
I am very surprised by this, and the regression on my system is happening for all versions of Foam since the feature was introduced, which makes me think something has changed in the way vscode is handling those links.
Seems related to this even though it's been fixed: https://github.com/microsoft/vscode/issues/115812

**Comment by YAS01-M** [comment]
Yes , this happened also after the update of vs code studio , but before threre is no issue . I removed everything , started now with vs code insiders but nothing changed . Don't know if this issue happened with all the people who are using Foam or just many of us .
Another screenshot before clicking on some of the markdown notes  :
![issue](https://user-images.githubusercontent.com/84593166/119244418-75116800-bb68-11eb-95f8-25e5e496cb4f.JPG)

**Comment by Saskia-R** [comment]
Same phenomenon on linux (x64), visual studio code version 1.56.2, Foam 0.13.3. The theory about vscode handling changing sounds logical.

**Comment by riccardoferretti** [comment]
in Discord it's been reported that the problem was introduced in VS Code between version 1.55.2 and 1.56.2.
Downgrading to version 1.55.2 will solve the issue.
We still haven't figured out (or, really, deeply looked into) what is causing the issue in 1.56+ - any help there would be greatly appreciated!

**Comment by dewinterjack** [comment]
I have submitted a [new issue ](https://github.com/microsoft/vscode/issues/125595)

**Comment by caseychu** [comment]
[This issue](https://github.com/microsoft/vscode/issues/124678) seems related. It seems like the API for constructing links has changed to [`webview.asWebviewUri`](https://github.com/microsoft/vscode/blob/307cb32f306a527a30a5b1756e8bf548c141192c/src/vs/vscode.d.ts#L6703), and here would be the relevant place to change in Foam:
https://github.com/foambubble/foam/blob/bca9756e2be6d06f7f297099386338f88946a418/packages/foam-vscode/src/features/preview-navigation.ts#L53

**Comment by riccardoferretti** [comment]
@kevinslin from the Dendron project was kind enough to provide an additional pointer to address the problem, which uses a completely different approach, see https://github.com/microsoft/vscode/issues/125595#issuecomment-869863675.

To unblock this issue the `asWebviewUri` feels like the easiest approach, we need to confirm it works.

The command based approach (from Dendron) is also pretty solid and has some added benefits in terms of flexibility, but requires a bit more work and thinking (they are forking the preview extension, we are not)

**Comment by pderaaij** [comment]
I gave it a look yesterday, but I got stuck. All the examples I found are about custom webview panels which the Markdown Preview isn't. As such I wasn't able to figure out how to get a `WebView` that I could use to call `asWebviewUri`. Stopped working on it yesterday and wanted to give it another go later this week, but pointers are more than welcome.

**Comment by pderaaij** [comment]
I found a solution triggered by the comments in this issue: https://github.com/microsoft/vscode/issues/126402. 

Creating a PR as we speak.

**Comment by pderaaij** [comment]
@riccardoferretti This one can be closed!

**Comment by riccardoferretti** [comment]
🎉
